### ISSUE TITLE
[accessibility] Popup's Close button should have an accessible name and `role="button"`

### ISSUE THREAD

**Issue by Malvoz** [openlayers/openlayers#12308]
The popup's Close button (https://openlayers.org/en/latest/examples/popup.html):

![popup-button](https://user-images.githubusercontent.com/26493779/118116237-5255b580-b3ea-11eb-8d65-15226f4dc9a2.png)

- Is missing an accessible name, e.g. `aria-label="Close"`
- Is missing `role="button"` (or should be `<button type="button">`) as it is not intended to behave like a regular link

**Comment by Malvoz** [comment]
Accessibility issues don't go away this easily, Mr. stale bot.
